Jenkinsfile

```diff
@@ -19,115 +39,140 @@ properties([
+                // the default resolution when omitting `defaultBranch` is to `master`
+                // this is wrong in our case, so explicitly set `develop` as default
+                // TODO: does this also work for PRs with different base branch?
+                discoverGitReferenceBuild(defaultBranch: 'develop')
+                recordIssues skipBlames: true,
+                    tools: [
+                        checkStyle(pattern: '**/build/reports/checkstyle/*.xml'),
+                        spotBugs(pattern: '**/build/reports/spotbugs/main/*.xml', useRankAsPriority: true),
+                        pmdParser(pattern: '**/build/reports/pmd/*.xml')
+                    ]
+
+                recordIssues skipBlames: true,
+                    tool: taskScanner(includePattern: '**/*.java,**/*.groovy,**/*.gradle', \
+                                        lowTags: 'WIBNIF', normalTags: 'TODO', highTags: 'FIXME')
+            }
         }
-    }
 
-    stage('Integration Tests') {
-        try {
-            sh './gradlew --console=plain integrationTest'
-        } catch (err) {
-            // Currently, all MTE tests are broken after the migration to gestalt v7.
-            // 
-            // By catching the error here the failed test execution should not fail the stage,
-            // and thereby no longer fail the complete build.
-            //
-            // See https://github.com/MovingBlocks/Terasology/issues/4757
-        } finally {
+        stage('Documentation') {
+            steps {
+                sh './gradlew --console=plain javadoc'
+                script {
+                    if (fileExists("build/docs/javadoc/index.html")) {
+                        step([$class: 'JavadocArchiver', javadocDir: 'build/docs/javadoc', keepAll: false])
+                        recordIssues skipBlames: true, tool: javaDoc()
+                    }
+                }
+            }
+        }
 
-            // Gradle generates both a HTML report of the unit tests to `build/reports/tests/*` and XML reports
-            // to `build/test-results/*`.
-            // We need to upload the XML reports for visualization in Jenkins. 
-            //
-            // See https://docs.gradle.org/current/userguide/java_testing.html#test_reporting
-            junit testResults: '**/build/test-results/integrationTest/*.xml', allowEmptyResults: true, healthScaleFactor: 0.0
+        stage('Integration Tests') {
+            steps {
+                sh './gradlew --console=plain integrationTest'
+            }
+            post {
+                failure {
+                    // Currently, all MTE tests are broken after the migration to gestalt v7.
+                    // 
+                    // By catching the error here the failed test execution should not fail the stage,
+                    // and thereby no longer fail the complete build.
+                    //
+                    // See https://github.com/MovingBlocks/Terasology/issues/4757
```

Comment:
🚦 This is something we'll want to try to test.

I don't _expect_ an empty `failure {}` block here to alter the outcome one way or the other, but I am not sure.

Reply:
removed this, as Jenkins was complaining that a empty block was a syntax error